service/history/ndc/workflow_resetter.go

```diff
@@ -505,8 +505,9 @@ func (r *workflowResetterImpl) failInflightActivity(
 		switch ai.StartedEventId {
 		case common.EmptyEventID:
 			// activity not started, noop
-			// override the activity time to now
+			// override the scheduled activity time to now
 			ai.ScheduledTime = timestamppb.New(now)
+			ai.FirstScheduledTime = timestamppb.New(now)
```

Comment:
Hmm... why are you resetting time here?

Reply:
this is called from "resetWorfklow". Current running activity will fail, and will be restarted. I'm not sure if the same ActivityInfo will be reused, so it make sense to also reset "firstScheduledTime"